README.md

```diff
@@ -4,40 +4,182 @@
 [![Telegram](https://img.shields.io/badge/Telegram-2CA5E0?style=flat-square&logo=telegram&logoColor=dark)](https://t.me/nilfoundation)
 [![Twitter](https://img.shields.io/twitter/follow/nil_foundation)](https://twitter.com/nil_foundation)
 
+<!-- START doctoc generated TOC please keep comment here to allow auto update -->
+<!-- DON'T EDIT THIS SECTION, INSTEAD RE-RUN doctoc TO UPDATE -->
+**Table of Contents**
+
+- [Introduction](#introduction)
+- [Starting with a ready Docker image](#starting-with-a-ready-docker-image)
+- [Building the toolchain](#building-the-toolchain)
+  - [Cloning the repository](#cloning-the-repository)
+  - [Building with Docker](#building-with-docker)
+  - [Building on host](#building-on-host)
+    - [Dependencies](#dependencies)
+    - [Building](#building)
+- [Proof Market Beta Access](#proof-market-beta-access)
+- [Proof Market Interaction](#proof-market-interaction)
+  - [1. Prepare zkLLVM (for circuit developers)](#1-prepare-zkllvm-for-circuit-developers)
+  - [2. Circuit Generation/Publishing (for circuit developers)](#2-circuit-generationpublishing-for-circuit-developers)
+  - [3. Proof Market Request Creation](#3-proof-market-request-creation)
+  - [4. Wait for proposals](#4-wait-for-proposals)
+  - [5. Submit proposal](#5-submit-proposal)
+  - [6. Order Matching](#6-order-matching)
+  - [7. Proof Generation](#7-proof-generation)
+  - [8. Proof Submission](#8-proof-submission)
+  - [9. Get Proof](#9-get-proof)
+- [Common issues](#common-issues)
+  - [Compilation Errors](#compilation-errors)
+  - [Submodule management](#submodule-management)
+  - [Compilation errors for Proof generator](#compilation-errors-for-proof-generator)
+  - [macOS](#macos)
+- [Documentation](#documentation)
+  - [Support](#support)
+
+<!-- END doctoc generated TOC please keep comment here to allow auto update -->
+
+
 # Introduction
 
-This repository serves as a PoC of how a =nil; Foundation built [proof market](https://proof.market/) would operate.
-This repository also provides a set of scripts/tools required to participate in the proof market.
+This repository also provides a set of scripts/tools required to participate in the
```

Comment:
> also provides 

"Also" to what? There is not any information before. 
Not a blocker

Reply:
Indeed )